packages/main/src/plugin/telemetry/telemetry-api.ts

```diff
@@ -0,0 +1,36 @@
+/**********************************************************************
+ * Copyright (C) 2024 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ ***********************************************************************/
+
+/**
+ * Frequency of collection:
+ *  dailyPerInstance - log the first time the event happens after startup.
+ *
+ * Enum to allow for future expansion, e.g. 'daily' or 'weekly'.
+ */
+export type Frequency = 'dailyPerInstance';
+
+export interface TelemetryRule {
+  // regex matching telemetry event
+  event: string;
```

Comment:
If it is a regexp why it's not defined as a regexp ?

Reply:
The interface matches what's in the json file to make for easy parsing, and needs to stay as a string since that's what you define there. I've created a map in latest commit to create and cache the RegExp at load time.